Correct the tooltip text for the toolbar X icon to be strictly correct, 'display-number:screen' should be ':display-number.screen'.
Also for the default window title.
Adjust the style of the Windows title in XDMCP mode from 'Xming - hostname' to 'hostname:display-number.screen'.


Copyright (C) Colin Harrison 2005-2008
http://www.straightrunning.com/XmingNotes/
http://sourceforge.net/projects/xming/ 

---
 xserver/hw/xwin/wincreatewnd.c |    8 ++++++--
 xserver/hw/xwin/wintrayicon.c  |    2 +-
 xserver/hw/xwin/winwindow.h    |    4 ++--
 3 files changed, 9 insertions(+), 5 deletions(-)

Index: xorg-server-1.5.2/xserver/hw/xwin/wintrayicon.c
===================================================================
--- xorg-server-1.5.2.orig/xserver/hw/xwin/wintrayicon.c
+++ xorg-server-1.5.2/xserver/hw/xwin/wintrayicon.c
@@ -59,7 +59,7 @@ winInitNotifyIcon (winPrivScreenPtr pScr
   /* Set display and screen-specific tooltip text */
   snprintf (nid.szTip,
 	    sizeof (nid.szTip),
-	    PROJECT_NAME " Server - %s:%d",
+	    PROJECT_NAME " Server:%s.%d",
 	    display, 
 	    (int) pScreenInfo->dwScreen);
 
Index: xorg-server-1.5.2/xserver/hw/xwin/winwindow.h
===================================================================
--- xorg-server-1.5.2.orig/xserver/hw/xwin/winwindow.h
+++ xorg-server-1.5.2/xserver/hw/xwin/winwindow.h
@@ -42,8 +42,8 @@
 #  define PROJECT_NAME		"Cygwin/X"
 #endif
 #define WINDOW_CLASS		"cygwin/x"
-#define WINDOW_TITLE		PROJECT_NAME " - %s:%d"
-#define WINDOW_TITLE_XDMCP	PROJECT_NAME " - %s"
+#define WINDOW_TITLE		PROJECT_NAME ":%s.%d"
+#define WINDOW_TITLE_XDMCP	"%s:%s.%d"
 #define WIN_SCR_PROP		"cyg_screen_prop rl"
 #define WINDOW_CLASS_X		"cygwin/x X rl"
 #define WINDOW_TITLE_X		PROJECT_NAME " X"
Index: xorg-server-1.5.2/xserver/hw/xwin/wincreatewnd.c
===================================================================
--- xorg-server-1.5.2.orig/xserver/hw/xwin/wincreatewnd.c
+++ xorg-server-1.5.2/xserver/hw/xwin/wincreatewnd.c
@@ -88,7 +88,9 @@ winCreateBoundingWindowFullScreen (Scree
     snprintf (szTitle,
 	    sizeof (szTitle),
 	    WINDOW_TITLE_XDMCP,
-	    g_pszQueryHost); 
+	    g_pszQueryHost,
+	    display,
+	    (int) pScreenInfo->dwScreen);
   else    
     snprintf (szTitle,
 	    sizeof (szTitle),
@@ -331,7 +333,9 @@ winCreateBoundingWindowWindowed (ScreenP
     snprintf (szTitle,
 	    sizeof (szTitle),
 	    WINDOW_TITLE_XDMCP,
-	    g_pszQueryHost); 
+	    g_pszQueryHost,
+	    display,
+	    (int) pScreenInfo->dwScreen);
   else    
     snprintf (szTitle,
 	    sizeof (szTitle),
